version/manager/template/template.go

```diff
@@ -0,0 +1,82 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the License);
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an AS IS BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/insolar/insolar/log"
+	"gopkg.in/yaml.v2"
+)
+
+type SFeature struct {
+	StartVersion string `yaml:"startversion"`
+	Description  string `yaml:"description"`
+}
+
+type SVersionTable struct {
+	V map[string]SFeature `yaml:"versiontable"`
+}
+
+//go:generate go run template.go
+func main() {
+	vtContent, _ := ioutil.ReadFile(path.Join("..", "..", "..", "versiontable.yml"))
+	buffer, err := parseYaml(vtContent)
+	if err != nil {
+		log.Error(err)
+	}
+	if err := ioutil.WriteFile(path.Join("..", "versiontable.go"), buffer.Bytes(), 0644); err != nil {
+		log.Error(err)
+	}
+}
+
+func parseYaml(vtContent []byte) (*bytes.Buffer, error) {
```

Comment:
in `logicrunner/golpugin/preprocessor` we use templates for codegenration, should we do the same here?

Reply:
I propose to think about this in the next stages.